src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java

```diff
@@ -153,4 +164,17 @@ boolean injectUser(RestRequest request) {
         return true;
 
     }
+
+    User injectUser(TransportRequest transportRequest, Task task, String action) {
+        String injectedUserString = threadPool.getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER_HEADER);
+        User user = getUser(injectedUserString);
+        if(user == null) {
+            return null;
+        }
+        auditLog.logSucceededLogin(user.getName(), true, null, transportRequest, action, task);
+        if (log.isTraceEnabled()) {
+            log.trace("Injected user object:{} ", user.toString());
+        }
+        return user;
```

Comment:
The volume of transport requests is quite high when compared to rest requests. Would this have any negative impact on audit log volume?

Also is there a chance that could result in repeated injections? (REST injects and the corresponding transport request also results in injection)

Reply:
This should be the first transport request and after this it should not trigger multiple injections.